fegrow/receptor.py

```diff
@@ -93,6 +93,7 @@ def optimise_in_receptor(
     relative_permittivity: float = 4,
     water_model: str = "tip3p.xml",
     platform_name: str = "CPU",
+    ligand_freeze: list[int] = None,
```

Comment:
Do you think something like `ligand_indices_to_freeze` might be clearer (if a bit long-winded)?

Also, should this be type-hinted as `Optional[list[int]]` (if you want to keep supporting Python < 3.10, or `list[int] | None` if not? Given that 3.10 has been out since 2021, would it make sense to enforce `python>=3.10` rather than 3.8, and maybe to run the CI on all supported versions rather than just 3.10 and 3.11? That's probably out of the scope of this review though, so feel free to ignore.

Reply:
Sounds all good. I used `Optional` until we drop the old Pythons. Thanks!